dimod/generators/fcl.py

```diff
@@ -122,17 +134,16 @@ def frustrated_loop(graph: GraphLike,
             continue
         good_cycles += 1
 
-        # If its a good cycle, modify J with it, according to plated solution
-        cycle_J = {}
-        for i, c in enumerate(cycle):
-            u, v = cycle[i - 1], cycle[i]
-            J = -planted_solution[u]*planted_solution[v]
-            cycle_J[(u, v)] = J
-
-        # randomly select an edge and flip it
-        idx = r.randint(len(cycle))
-        cycle_J[(cycle[idx - 1], cycle[idx])] *= -1
-
+        if plant_solution:
+            # randomly select from all frustrated loops with one AFM edge
+            cycle_J = {(cycle[i - 1], cycle[i]) : -1 for i in range(len(cycle))}
+            idx = r.randint(len(cycle))
+            cycle_J[(cycle[idx - 1], cycle[idx])] = 1
+        else:
+            # randomly select from all frustrated loops (odd number of AFM edges)
+            cycle_J = {(cycle[i], cycle[i+1]) : -1 for i in range(len(cycle)-1)}
+            cycle_J[(cycle[-1],cycle[0])] = (1 - 2*(len(cycle_J) & 1))*math.prod(cycle_J.values())
```

Comment:
unfortunately [math.prod](https://docs.python.org/3/library/math.html#math.prod) only came in in Python 3.8. You could use [np.prod](https://numpy.org/doc/stable/reference/generated/numpy.prod.html) instead.